### DISCUSSION TITLE
i18n translation und formatting archive pages listing

### DISCUSSION THREAD

**Discussion by rainerrose** [razonyang/hugo-theme-bootstrap#998] [Q&A] [answered]
I still have a problem in the display of the date format in the **archive** pages in the right localization.

Currently the date is displayed in english language in the default format defined in line 23 [themes/hugo-theme-bootstrap/layouts/partials/archives/list.html](https://github.com/razonyang/hugo-theme-bootstrap/blob/master/layouts/partials/archives/list.html#L23 )

~~~
2023
---
16 Beiträge insgesamt
[Sonstiges]         Oct 30
[GIT]               Oct 11
[Discord-Bot]       Sep 21
~~~

Probably did not understand the configuration of the theme.

My current config:
~~~ toml
[de]
  languageName = "Deutsch"
  weight = 1
  defaultContentLanguage = "de"
  [de.params]
    dateFormat = ':date_full'
    description = "Anleitungen, HowTos, Code-Schnipsel & Blog"
    pygmentsUseClasses = true
    comment = false
  [de.params.archive]
    languageName = "Deutsch"
    defaultContentLanguage = "de"
    dateFormat = "1. Jan"
    description = "Das Archiv der Homepage"
  
[en]
  languageName = "English"
  weight = 2
  [en.params]
    pygmentsUseClasses = true
    comment = false

[fr]
  languageName = "Français"
  weight = 3
  [fr.params]
    dateFormat = "02-01-2006"
    comment = false
[blackfriday.params]
  hrefTargetBlank = true
~~~

I thought that the section `[de.params.archive]` was responsible for displaying the right format in the german language.

~~~ toml
  [de.params.archive]
    languageName = "Deutsch"
    defaultContentLanguage = "de"
    dateFormat = "1. Jan"
~~~

The expected output would be

~~~
2023
---
16 Beiträge insgesamt
[Sonstiges]         Okt 30
[GIT]               Okt 11
[Discord-Bot]       Sep 21
~~~

`Oct` have changed to `Okt`

But no matter what I enter there, unfortunately nothing changes.

for example
~~~ toml
dateFormat = "02-01-2006"
~~~

its still ` Oct 30` displayed.

Could someone give me a hint as to what I am doing wrong?

**Comment by razonyang** [comment, marked as answer]
It's a bug, the theme doesn't read the parameter correctly, now it was fixed, you can use the latest changes by upgrade to the `master` branch.

Btw, the `languageName` and `defaultContentLanguage` and `description` are unused/invalid below the `[de.params.archive]` block.

```
  [de.params.archive]
    dateFormat = ":date_medium"
```

See also:

- https://gohugo.io/functions/time/format/#localization
- https://hbs.razonyang.com/v1/en/docs/getting-started/upgradation/

Thanks for the feedback.

**Comment by rainerrose** [comment]
Oh dear. I wish I had posted this issue sooner. I've been tinkering with it for a few weeks now. 
Many thanks for the - once again - quick response. I think I'll give it a try at the weekend and mark it as a solution. 
 
> Btw, the languageName and defaultContentLanguage and description

Yes, correct. Thanks for the reminder. That was a desperate attempt from my debugging. I will remove that.

> Thanks for the feedback.

You are welcome. Thanks for the bugfixing.

**Comment by rainerrose** [comment]
Hello @razonyang , here I am again. 
I have pulled the repo and implemented the suggested changes. It works wonderfully on `http://localhost:1313/archives/`.
Unfortunately, this does not affect subpages, e.g. `http://localhost:1313/archives/2023/` there is again the above mentioned problem. 

Here is again the default english and default dateFormat in the lines with the page title and the month paragraphs.

The error should be in this file:

`hugo-theme-bootstrap/layouts/partials/archives/year.html`

I will open an issue for this in a moment.
